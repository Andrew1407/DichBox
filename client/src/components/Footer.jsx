import React from 'react';
import '../styles/footer.css';

const Footer = () => (
  <div id="footer" data-testid="footer-test">
    <p id="version">
      <small>
       <b>version: 0.5.6</b>
      </small>
      </p>
    <p>
      <b>DichES</b>
    </p>  
    <p>
      <b>
        <small>
          <a id="repo-ref" href="https://github.com/Andrew1407/DichBox" target="_blank">git repo</a>
        </small>
      </b>
    </p>
  </div>
);

export default Footer;
