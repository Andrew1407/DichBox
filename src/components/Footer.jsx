import React from 'react';
import '../styles/footer.css';

const Footer = () => (
  <div id="footer">
    <p id="version">
      <small>
       <b>version: 0.2.1</b>
      </small>
      </p>
    <p>
      <b>DichES</b>
    </p>  
    <p>
      <b>
        <small>
          <a id="repo-ref" href="https://github.com/Andrew1407/DichBox" target="_blank">git repo</a>
        </small>
      </b>
    </p>
  </div>
);

export default Footer;
