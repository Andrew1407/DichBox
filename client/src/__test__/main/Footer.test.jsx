import React from 'react';
import { render } from '@testing-library/react';
import Footer from '../../components/Footer';

describe('Footer tests', () => {
  const { getByTestId } = render(<Footer />);
  const foundFooter = getByTestId('footer-test');

  const tests = {
    'has version': () => {
      const version = /version: \d\.\d\.\d/;
      expect(foundFooter).toHaveTextContent(version);
    },
    'has title': () => {
      expect(foundFooter).toHaveTextContent('DichES');
    },
    'has href': () => {
      const [ srcRef ] = Array.from(foundFooter.getElementsByTagName('a'));
      expect(foundFooter).toHaveTextContent('git repo');
      expect(srcRef).toHaveAttribute('href', 'https://github.com/Andrew1407/DichBox');
      expect(srcRef).toHaveAttribute('target', '_blank');
    }
  };

  for (const key in tests)
    it(key, tests[key]);
});
