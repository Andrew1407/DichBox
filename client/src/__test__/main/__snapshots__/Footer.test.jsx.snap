// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer tests matches snapshot 1`] = `
<div
  data-testid="footer-test"
  id="footer"
>
  <p
    id="version"
  >
    <small>
      <b>
        version: 0.6.2
      </b>
    </small>
  </p>
  <p>
    <b>
      DichES
    </b>
  </p>
  <p>
    <b>
      <small>
        <a
          href="https://github.com/Andrew1407/DichBox"
          id="repo-ref"
          target="_blank"
        >
          git repo
        </a>
      </small>
    </b>
  </p>
</div>
`;
